src/gc.c

```diff
@@ -1670,10 +1711,48 @@ static void gc_sweep_pool(int sweep_full)
             p->newpages = NULL;
         }
     }
+        // wait for concurrent sweep to finish
+    if (jl_n_gcthreads != 0) {
+        while (jl_atomic_load(&gc_concurrent_sweep_running)) {
+            jl_cpu_pause();
+        }
+    }
+    gc_sweep_queue.current = gc_sweep_queue.begin;
+
+    sweep_pool_pagetable(sweep_full);
+
+    // wake up GC threads
+    uv_mutex_lock(&gc_threads_lock);
+    jl_atomic_store(&gc_sweep_mask, UINT64_MAX);
+    uv_cond_broadcast(&gc_threads_cond);
+    uv_mutex_unlock(&gc_threads_lock);
 
     // the actual sweeping
-    sweep_pool_pagetable(pfl, sweep_full);
+    gc_sweep_pool_parallel(NULL, 1);
+    for (int i = 1; i <= jl_n_gcthreads; i++) {
+        while (jl_atomic_load(&gc_sweep_mask) & (1ull << i)) {
```

Comment:
Why the for-loop? Isn't this equivalent to:

```
while (jl_atomic_load(&gc_sweep_mask) > 0)
   jl_cpu_pause();
```

Reply:
These are not equivalent if the number of GC threads is less than 63. The ith GC thread resets the ith bit of the `gc_sweep_mask`, so the upper bits of it may be left untouched even after all threads have swept.

Reply:
We should replace this by a bitmap to avoid limiting the number of GC threads to 63 though.